pkg/controller/storageprofile-controller.go

```diff
@@ -199,6 +204,25 @@ func (r *StorageProfileReconciler) deleteStorageProfile(name string, log logr.Lo
 	return r.checkIncompleteProfiles()
 }
 
+func removeStorageClass(name string) {
+	_, found := scProvisionerMap[name]
+	if found {
+		delete(scProvisionerMap, name)
+	}
+}
+
+func addStorageClass(sc *storagev1.StorageClass) {
+	scProvisionerMap[sc.Name] = sc.Provisioner
+}
+
+func isNoProvisioner(name string) bool {
+	provisioner, found := scProvisionerMap[name]
+	if found && provisioner == "kubernetes.io/no-provisioner" {
```

Comment:
Should we use a constant for the no-provisioner string?

Reply:
This ties closely to another discussion on this PR:
https://github.com/kubevirt/containerized-data-importer/pull/2226#discussion_r843077393

How expensive would it be to bring in https://pkg.go.dev/k8s.io/kubernetes/pkg/controller/volume/persistentvolume/util?

Reply:
if I try to do that, I'm going to end up calling it NotSupportedProvisioner which is kinda confusing in our context, because it is supported. Perhaps it's better to leave it as-is.